hal/source/arg_pack.cpp

```diff
@@ -30,6 +31,10 @@ namespace util {
 /// @{
 
 namespace {
+// http://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
+// This will round up to power of 2 for all values except 0, which will
+// result in 0. This is good enough for our purposes where it is based on
+// a non-zero size value.
```

Comment:
If a value of `0x8000000000000001` or greater is passed in, the result will be `0`.

Reply:
fair enough I can reflect in the comment.

Reply:
`0x8000000000000000` is 2^63, not 2^32. Values from 2^32 to 2^63 should be handled correctly, I think.

Reply:
you are right, fixed.